File: Domain/Entities/Customer.cs
using System;
using System.Collections.Generic;

namespace Queue.Domain.Entities;

public partial class Customer
{
    public int Id { get; set; }

    public Guid Guid { get; set; }

    public int ShopId { get; set; }

    public int? UserId { get; set; }

    public string Name { get; set; } = null!;

    public string Phone { get; set; } = null!;

    public string? Email { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public int? CreatedBy { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public int? UpdatedBy { get; set; }

    public virtual ICollection<CustomerNote> CustomerNotes { get; set; } = new List<CustomerNote>();

    public virtual ICollection<CustomerTagMap> CustomerTagMaps { get; set; } = new List<CustomerTagMap>();

    public virtual Shop Shop { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: Frontend/src/types/shop/customer.ts
export type CustomerTag = {
    id: number
    name: string
    isActive: boolean
    createdAt: string
    createdBy?: number
    updatedAt?: string
    updatedBy?: number
}

export type CustomerNoteType = {
    id?: number
    note: string
    createdAt?: string
    createdBy?: number | string
}

export type CustomerType = {
    id: number
    shopId: number
    name: string
    phone: string
    isActive: boolean
    tags?: CustomerTag[]
    notes?: CustomerNoteType[]
    createdAt: string
    createdBy: string
}